### ISSUE TITLE
ModuleNotFoundError: No module named 'transformers.hf_api'

### ISSUE THREAD

**Issue by Pathos14489** [neuml/txtai#274]
I feel like I'm going crazy, but on two seperate distros (Xubuntu 22.04 and now PopOS 22.04) I'm running into an issue that I didn't have before reinstalling today.

If I install via `pip install txtai[all]` or `[pipeline]`, after having gotten the latest version installed, it downgrades my version. When txtai is downgraded to 3.5.0, I run into this error:
```
    from txtai.embeddings import Embeddings
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/embeddings/__init__.py", line 5, in <module>
    from .base import Embeddings
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/embeddings/base.py", line 12, in <module>
    from ..scoring import ScoringFactory
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/scoring/__init__.py", line 5, in <module>
    from .base import Scoring
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/scoring/base.py", line 10, in <module>
    from ..pipeline.tokenizer import Tokenizer
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/pipeline/__init__.py", line 20, in <module>
    from .translation import Translation
  File "/home/pathos/.local/lib/python3.10/site-packages/txtai/pipeline/translation.py", line 15, in <module>
    from transformers.hf_api import HfApi
ModuleNotFoundError: No module named 'transformers.hf_api'
```
And if I don't install this, and install the latest version with `pip install git+https://github.com/neuml/txtai` or by `git clone` and `pip install .`, this issue is fixed and I can run most of the txtai. HOWEVER, then when trying to use Textractor, I run into the follow error:
```
ImportError: Textractor pipeline is not available - install "pipeline" extra to enable
```
Which, going back to the beginning, if I install the pipeline extra, 4.5.0 gets uninstalled and 3.5.0 gets installed, and then the first issue crops up that I am unsure how to fix, because reinstalling transformers has done absolutely nothing on both distros now.

Also on the side, before reinstalling because of a bugged install(I tried to update PopOS 21.04 to 22.04 and I think it failed somewhere along the line for some reason, because it was very unstable afterwards and eventually failed entirely this morning.), it was working fantastically and I must say this is an incredible package.
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.68.02    Driver Version: 510.68.02    CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:2D:00.0  On |                  N/A |
|  0%   53C    P0    55W / 260W |   1259MiB / 11264MiB |      5%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```
`Python 3.10.4`

**Comment by Pathos14489** [comment]
After going through installing each of the dependencies one by one, I found that in Python 3.10.4 onnx, onnxruntime are incompatible at this time. And maybe fasttext. I had to fix it, but I can't remember why. But in the end, I noticed that my deployment server could still run it, and the only difference was Python 3.10.4 (The transformers issue also went away when running 3.5.0 instead of the latest version as well. 3.10.4 seems to break both the pypi version of txtai and the repo version for seperate reasons.)

**Comment by davidmezzetti** [comment]
Thank you for researching this further. You definitely do not want to use txtai 3.5, it's really out of date at this point.

onnx and onnxruntime both don't have Python 3.10 wheels on PyPI. onnx has a source distribution and can be built with the right OS deps installed. onnxruntime does not. This is an open issue as referenced below:

https://github.com/microsoft/onnxruntime/issues/9782
https://github.com/microsoft/onnxruntime/issues/11117
https://github.com/microsoft/onnxruntime/issues/11332

Assuming you don't need ONNX support, as a workaround you can do this:

```
pip install txtai[pipeline]
pip install txtai --upgrade
```

That will do what you had previously but then upgrade txtai back to the latest version. If Python 3.10.x isn't that important, you can run with Python 3.9 and everything will work.

Alternatively you can run `pip install txtai` and then manually install the packages you need from the pipeline extras list.

```
extras["pipeline"] = [
    "beautifulsoup4>=4.9.3",
    "fasttext>=0.9.2",
    "imagehash>=4.2.1",
    "nltk>=3.5",
    "onnx>=1.10.1",
    "onnxmltools>=1.9.1",
    "onnxruntime>=1.8.1",
    "pandas>=1.1.0",
    "pillow>=9.0.1",
    "sentencepiece>=0.1.91",
    "soundfile>=0.10.3.post1",
    "tika>=1.24",
    "timm>=0.4.12",
]
```

The onnxruntime project says this will be resolved with 1.12 but no date is set on that.

**Comment by xo-r32** [comment]
- [ ] **_[ ]()_**

**Comment by davidmezzetti** [comment]
This issue has been resolved upstream with [onnxruntime v1.12](https://github.com/microsoft/onnxruntime/releases/tag/v1.12.0)

**Comment by ivostoykov** [comment]
Hi All,

I crashed with the same problem on Ubuntu 22.04 (patched) in `venv`
```
$ python3 --version
Python 3.10.12
$ pip --version
pip 24.0 from ~/Projects/Python/txtai/lib/python3.10/site-packages/pip (python 3.10)
```
then I tried:

```
pip install txtai[pipeline]
pip install txtai[all]
```
always crashing with the same error:

```
Collecting fasttext>=0.9.2 (from txtai[all])
  Using cached fasttext-0.9.2.tar.gz (68 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      /Projects/Python/txtai/bin/python3: No module named pip
      Traceback (most recent call last):
        File "<string>", line 38, in __init__
      ModuleNotFoundError: No module named 'pybind11'
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Projects/Python/txtai/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Projects/Python/txtai/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Projects/Python/txtai/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-ehfcvlx_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-ehfcvlx_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-ehfcvlx_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-ehfcvlx_/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 72, in <module>
        File "<string>", line 41, in __init__
      RuntimeError: pybind11 install failed.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

```

I'll appreciate any help. Thanks
